### PR TITLE
Cinnamon 5.8.x

### PR DESCRIPTION
**Pull request by mhmdanas**
#### Testing the changes
- I tested the changes in this PR: **briefly** (I don't use Cinnamon but did briefly test the DE)

Testing from actual Cinnamon users would be appreciated.

### CODE REVIEW

File: srcpkgs/cjs/template [hunk 1]
```diff
@@ -1,26 +1,21 @@
...
-build_helper=gir
+build_helper="gir"
 # XXX: disable tests because they need dbus, X and GTK, same as gjs
 configure_args="-Dskip_dbus_tests=true -Dskip_gtk_tests=true
  -Dprofiler=disabled -Dinstalled_tests=false"
-hostmakedepends="pkg-config gobject-introspection glib-devel dbus mozjs78-devel"
-makedepends="dbus-glib-devel readline-devel gobject-introspection
- gtk+3-devel mozjs78 mozjs78-devel"
+hostmakedepends="pkg-config glib-devel dbus"
+makedepends="dbus-glib-devel readline-devel gtk+3-devel mozjs102
```
Comment (sgn) [inline srcpkgs/cjs/template:11]: We may need `gtk4-devel`

File: srcpkgs/cjs/patches/fix-configure.diff [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- a/installed-tests/js/meson.build	2023-06-19 19:51:58.989737570 +0100
++++ b/installed-tests/js/meson.build	2023-06-19 19:51:23.010310261 +0100
+@@ -203,7 +203,7 @@
+ # during build should be run using dbus-run-session
+ 
+ dbus_tests = ['GDBus']
+-if have_gtk4 and not get_option('skip_gtk_tests')
```
[the remark this answers is not in the document]
Comment (mhmdanas) [reply srcpkgs/cjs/patches/fix-configure.diff:7]: Without this, `do_configure` fails with this error:

```
installed-tests/js/meson.build:206:3: ERROR: Unknown variable "have_gtk4".
```

I also plan to submit a fix upstream.

File: srcpkgs/cjs/template (hunk 1 above)
Comment (mhmdanas) [reply srcpkgs/cjs/template:11]: Reading through meson.build, it seems that gtk4-devel is only needed for tests (and gtk+3-devel doesn't seem to be needed at all).

**Comment by dmarto** [comment]
@mhmdanas should I test this (I daily cinnamon), or should I wait for non-draft status?

**Comment by mhmdanas** [comment]
@dmarto this should be fine to test. I've marked it as draft because of the lack of testing and because of the CI check failure on cjs.

**Comment by dmarto** [comment]
Rebased on 14cd86c79b2e1efa0da0d5ad1964def54080c439; built and running for close to an hour already, and will continue to use it.

My only request is to drop "touchegg" from `depends` as it is an optional dependency, and as one of the crazy people that daily uses a touchpad on a computer (not laptop), I also don't find it particularly good.

**Comment by mhmdanas** [comment]
Thanks for the feedback.

> My only request is to drop "touchegg" from depends as it is an optional dependency, and as one of the crazy people that daily uses a touchpad on a computer (not laptop), I also don't find it particularly good.

Do you think adding it to `cinnamon-all` is a good idea? I added it because it was recommended in one of the package's changelogs (though I don't recall which one).

**Comment by dmarto** [comment]
hah, I didn't know about `cinnamon-all`, I don't have it, but I guess it makes sense to include the dep there

**Comment by mhmdanas** [comment]
@dmarto I pushed some more updates, can you please test those too?

File: srcpkgs/cjs/patches/fix-configure.diff (hunk 2 above)
[the remark this answers is not in the document]
Comment (mhmdanas) [reply srcpkgs/cjs/patches/fix-configure.diff]: A fix was implemented upstream (https://github.com/linuxmint/cjs/commit/fb480aae4eff422c3acd275c0108e970eafcdcb9), and I replaced my patch with it.

**Comment by dmarto** [comment]
idk, what you changed, so here is what xbps update based on https://github.com/void-linux/void-packages/commit/138b1fc065f1d64c75fc7ad97c557153d0375d04
```
Name                       Action    Version           New version            Download size
cinnamon                   update    5.8.3_1           5.8.4_1                - 
cinnamon-control-center    update    5.8.1_1           5.8.2_1                - 
cinnamon-screensaver       update    5.8.0_1           5.8.1_1                - 
cinnamon-translations      update    5.8.1_1           5.8.2_1                - 
libcinnamon-control-center update    5.8.1_1           5.8.2_1                - 
muffin                     update    5.8.0_1           5.8.1_1                - 
```

will report back, in a day :)

**Comment by mhmdanas** [comment]
@dmarto thanks for testing. What remains at this point is fixing the `cjs` tests on musl; I opened [a ticket](https://github.com/linuxmint/cjs/issues/118) for that.

File: srcpkgs/cjs/template [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
-# but some tests will fail.
-make_check=extended
+checksum=04e7c10a0a4d88ba446e6fb4f4788f732984e0318d775c646634a8619dfff7d2
+make_check_pre="xvfb-run"
+
+if [ "$XBPS_CHECK_PKGS" ]; then
+	configure_args+=" -Dskip_dbus_tests=false -Dskip_gtk_tests=false"
+else
+	configure_args+=" -Dskip_dbus_tests=true -Dskip_gtk_tests=true"
+fi
```
Comment (sgn) [inline srcpkgs/cjs/template:24]: Do not skip anything. Just build them all.

Comment (mhmdanas) [reply srcpkgs/cjs/template]: Without them, buliding without tests fails with a `meson.build:272:19: ERROR: Program 'dbus-run-session' not found or not executable` error as can be seen in the CI, so I think they ought to be kept (at least the `-Dskip_dbus_tests=true`, I haven't figured out if the gtk parameter is needed too). The gjs template also does a similar thing.

Comment (sgn) [reply srcpkgs/cjs/template]: I meant add those `checkdepends` into `hostmakedepends`

File: srcpkgs/nemo-compare/template [hunk 4]
```diff
@@ -11,5 +11,6 @@ short_desc="Nemo compare extention"
 maintainer="Orphaned <orphan@voidlinux.org>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/linuxmint/nemo-extensions"
-distfiles="https://github.com/linuxmint/nemo-extensions/archive/refs/tags/${version}.tar.gz"
-checksum=6a0b83e6a9129c7217c29fbc992109d8f6eedbe8f076da863c507c0005182404
+# distfiles="https://github.com/linuxmint/nemo-extensions/archive/refs/tags/${version}.tar.gz"
```
Comment (classabbyamp) [inline srcpkgs/nemo-compare/template:14]: is this needed?

Comment (mhmdanas) [reply srcpkgs/nemo-compare/template:14]: For 5.8.1, yeah, see https://github.com/linuxmint/nemo-extensions/tags.

(If you mean why use 5.8.1, it has this one fix: https://github.com/linuxmint/nemo-extensions/commit/d25b09fcd00c4970c70485fb50d3595d8e6c8ef8)

**Pull request merged**